clients/cobol-lsp-vscode-extension/src/ProfileService.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2020 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Broadcom, Inc. - initial API and implementation
+ */
+
+import { IProfile } from "@zowe/imperative";
+import * as path from "path";
+import * as vscode from "vscode";
+import { SETTINGS_SECTION } from "./extension";
+import { ProfilesMap, ZoweApi } from "./ZoweApi";
+
+export class ProfileService {
+    constructor(private zoweApi: ZoweApi) {
+    }
+
+    public async getProfile(depFile: vscode.Uri): Promise<string> {
+        const programName = this.getProgramNameFromDepFile(depFile);
+        const detectedProfile = (await this.findProfileByDependenciesFile(depFile))
+            || this.tryGetProfileFromSettings(programName);
+        if (detectedProfile) {
+            return detectedProfile;
+        }
+        const profiles: ProfilesMap = await this.zoweApi.listZOSMFProfiles();
+        if (Object.keys(profiles).length === 0) {
+            await vscode.window.showErrorMessage("Zowe profile is missing.");
+            return undefined;
+        }
+        if (Object.keys(profiles).length === 1) {
+            return Object.keys(profiles)[0];
+        }
+        const defaultName = this.zoweApi.getDefaultProfileName();
+        let items: vscode.QuickPickItem[] = [];
+        Object.keys(profiles).forEach(name => {
+            const profile: IProfile = profiles[name];
+            const item: vscode.QuickPickItem = {
+                description: profile.username + "@" + profile.host + ":" + profile.port,
+                label: name,
+            };
+            if (defaultName === name) {
+                items = [item].concat(items);
+            } else {
+                items.push(item);
+            }
+        });
+
+        const selectedProfile = await vscode.window.showQuickPick(items,
+            { placeHolder: defaultName, canPickMany: false });
+        if (selectedProfile) {
+            let settingsProfiles = vscode.workspace.getConfiguration().get(SETTINGS_SECTION + ".profiles");
+            settingsProfiles = settingsProfiles ? settingsProfiles : {};
```

Comment:
you don't get any error here? can you assign empty object to the settingsProfile?

Reply:
It is an empty object according to package.json
I use it as a map to store a profile name for program.